src/analyze/analyze.c

```diff
@@ -2145,7 +2179,56 @@ static int do_condition(int argc, char *argv[], void *userdata) {
 }
 
 static int do_verify(int argc, char *argv[], void *userdata) {
-        return verify_units(strv_skip(argv, 1), arg_scope, arg_man, arg_generators);
+        char **filename;
+        _cleanup_strv_free_ char **filenames = NULL;
+        _cleanup_(rm_rf_and_freep) char *tempdir = NULL;
+        bool setup_alias = false;
+
+        STRV_FOREACH(filename, strv_skip(argv, 1))
+                if (strchr(*filename, ':')) {
+                        setup_alias = true;
+                        break;
+                }
+
+        if (setup_alias) {
+                int r, i = 1;
+                char *dst;
+
+                r = setup_working_dir(&tempdir);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to setup working directory");
+
+                STRV_FOREACH(filename, strv_skip(argv, 1)) {
+                        if (strchr(*filename, ':')) {
+                                _cleanup_free_ char *src = NULL, *argname = NULL;
+                                struct stat statbuf;
+
+                                argname = strdup(*filename);
+                                r = extract_first_word((const char **)filename, &src, ":", 0);
+                                if (r < 0 || !*filename)
+                                        return log_error_errno(r, "Failed to parse alias %s", argname);
+
+                                dst = path_join(tempdir, basename(*filename));
+                                if (!dst)
+                                        return log_oom();
+
+                                r = stat(src, &statbuf);
+                                if (r < 0)
+                                        return log_error_errno(errno, "Couldn't stat %s", src);
+
+                                r = copy_file(src, dst, 0, statbuf.st_mode, 0, 0, COPY_REFLINK);
+                                if (r < 0)
+                                        return log_error_errno(r, "Couldn't copy %s to alias %s", src, dst);
+
+                                strv_push(&filenames, dst);
+                        } else
+                                strv_push(&filenames, strdup(argv[i]));
```

Comment:
`strdup` might fail - but instead, just use `strv_extend(.. argv[i])` which does the strdup and checks for errors, etc